tests/python/test_python_frontend.py

```diff
@@ -4342,7 +4342,75 @@ def fusion_func(fd: FusionDefinition):
+        L = 64
+        S = 128
+        E = 64
 
+        inputs = [
+            torch.randn((N, H, L, E), dtype=torch.bfloat16, device="cuda:0"),
+            torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0"),
+            torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0"),
+        ]
+
+        def fusion_func(fd: FusionDefinition):
+            q = fd.define_tensor(shape = [-1, -1, -1, -1], contiguity = [True, True, True, True], dtype=DataType.BFloat16, is_cpu=False)
+            k = fd.define_tensor(shape = [-1, -1, -1, -1], contiguity = [True, True, True, True], dtype=DataType.BFloat16, is_cpu=False)
+            v = fd.define_tensor(shape = [-1, -1, -1, -1], contiguity = [True, True, True, True], dtype=DataType.BFloat16, is_cpu=False)
+            attn, *intermediate_results = fd.ops.sdpfa_fwd(q, k, v)
+            fd.add_output(attn)
+        nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
+        torch.testing.assert_close(nvf_out[0], F.scaled_dot_product_attention(*inputs))
+    
+    def test_sdpa_bwd(self):
+        N = 16
+        H = 32
+        L = 64
+        S = 128
+        E = 64
+
+        grad_output = torch.randn((N, H, L, E), dtype=torch.bfloat16, device="cuda:0")
+        q = torch.randn((N, H, L, E), dtype=torch.bfloat16, device="cuda:0")
+        k = torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0")
+        v = torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0")
+
+        dropout_p = 0.0
+        is_causal = False
+        output, log_sumexp, cum_seq_q, cum_seq_q, query_seq_len, key_seq_len, philox_seed, philox_offset, _ = torch.ops.aten._scaled_dot_product_flash_attention(
+            q, k, v, dropout_p, is_causal, return_debug_mask=False, scale = 1 / E**0.5
+        )
+
+        ref_grad = torch.ops.aten._scaled_dot_product_flash_attention(
+            q, k, v, dropout_p, is_causal, return_debug_mask=False, scale = 1 / E**0.5
+        )
+        query_seq_tensor = torch.scalar_tensor(query_seq_len, dtype=torch.int64)
+        key_seq_tensor = torch.scalar_tensor(query_seq_len, dtype=torch.int64)
+
+        def fusion_func(fd: FusionDefinition, inputs:list):
+            grad_output = fd.from_pytorch(inputs[0])
+            q = fd.from_pytorch(inputs[1])
+            k = fd.from_pytorch(inputs[2])
+            v = fd.from_pytorch(inputs[3])
+            output = fd.from_pytorch(inputs[4])
+            log_sumexp = fd.from_pytorch(inputs[5])
+            query_seq_len = fd.from_pytorch(inputs[6])
+            key_seq_len = fd.from_pytorch(inputs[7])
+            philox_seed = fd.from_pytorch(inputs[8])
+            philox_offset = fd.from_pytorch(inputs[9])
+            dropout_p = fd.define_scalar(0.0)
+            is_causal = fd.define_scalar(False, dtype=DataType.Bool)
+
+            grad_query, grad_key, grad_value= fd.ops.sdpfa_bwd(grad_output, q, k, v, output, log_sumexp, query_seq_len, key_seq_len, dropout_p, is_causal, philox_seed, philox_offset)
```

Comment:
@rdspring1: Here, using `fd.ops.sdpfa_bwd` does not allow `dropout_p, is_causal` to be None.

Note, that in `SdpaBwdOpRecord`, I skip printing `None` arguments. If I allow these arguments to be printed I run into the following error:
```
FAILED tests/python/test_python_frontend.py::TestNvFuserFrontend::test_sdpa_bwd - NameError: name 'None0' is not defined
```